### PR TITLE
BIGTOP-3819. Adding Puppet Recipes for Apache Phoenix

### PR DESCRIPTION
**Pull request by vivostar**
### Description of PR
Adding Puppet Recipes for Apache Phoenix

### How was this patch tested?
```shell
$ cd provisioner/docker
$ ./docker-hadoop.sh -d \
       --create 3 \
       --image bigtop/puppet:trunk-centos-7 \
       --memory 8g \
       --repo http://repos.bigtop.apache.org/releases/3.1.0/centos/7/\$basearch \
       --disable-gpg-check \
       --stack hdfs,yarn,mapreduce,hbase,phoenix
```

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [x] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

File: bigtop-deploy/puppet/modules/phoenix/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    package { 'phoenix':
+      ensure => latest,
+      require => Package["hbase"],
+    }
+    file { 'phoenix-server-lib': 
+      path => '/usr/lib/hbase/lib/phoenix-server.jar',
+      target => '/usr/lib/phoenix/phoenix-server.jar',
+      ensure => link,
+      require => Package['phoenix'],
+    }
```
Comment (iwasakims) [inline bigtop-deploy/puppet/modules/phoenix/manifests/init.pp:34]: How about adding `notify` to restart hbase-master and hbase-regionserver after the link is created?

Comment (vivostar) [reply bigtop-deploy/puppet/modules/phoenix/manifests/init.pp:34]: hi, @iwasakims, thanks for you advice. only there are some file changes, the relationship metaparameter **notify** will notify target resource to refresh. However, the phoenix-server-lib file resource operate a jar file, so we seldom edit it as we edit a text file or config file, the file maybe never changes, and then never refresh target resource(restart service). we can use **exec** resource file to implement the **restart** tasks.

**Comment by sekikn** [comment]
Thank you for this improvement @vivostar!
I tried this PR on Ubuntu 18.04 and came across the following error. Would you take a look? systemctl seems to exist in /bin on that Docker image.

```
$ ./docker-hadoop.sh -d -C config_ubuntu-18.04.yaml -r file:///bigtop-home/output/apt -G -k zookeeper,hdfs,hbase,phoenix -c 3

...

Notice: /Stage[main]/Phoenix::Server/Package[phoenix]/ensure: created
Notice: /Stage[main]/Phoenix::Server/File[phoenix-server-lib]/ensure: created
Error: Could not find command 'systemctl'
Error: /Stage[main]/Phoenix::Hbase_server_restart/Exec[hbase_server_restart]/returns: change from 'notrun' to ['0'] failed: Could not find command 'systemctl'

...

Notice: /Stage[main]/Hadoop_hbase::Master/Package[hbase-master]/ensure: created
Error: Could not find command 'systemctl'
Error: /Stage[main]/Phoenix::Hbase_master_restart/Exec[hbase_master_restart]/returns: change from 'notrun' to ['0'] failed: Could not find command 'systemctl'
```

**Comment by vivostar** [comment]
tested on ubuntu-18.04
```shell
./docker-hadoop.sh -d \
       --create 1 \
       --image bigtop/puppet:trunk-ubuntu-18.04 \
       --memory 8g \
       --repo http://repos.bigtop.apache.org/releases/3.1.0/ubuntu/18.04/\$\(ARCH\)\
       --disable-gpg-check \
       --stack hdfs,yarn,mapreduce,hbase,phoenix
./docker-hadoop.sh -e 1 /bin/bash

/# python3 /usr/lib/phoenix/bin/sqlline.py localhost
...
0: jdbc:phoenix:localhost> CREATE TABLE t(c VARCHAR PRIMARY KEY);
No rows affected (1.218 seconds)
0: jdbc:phoenix:localhost> UPSERT INTO t VALUES ('foo');
1 row affected (0.085 seconds)
0: jdbc:phoenix:localhost> SELECT * FROM t;
+-----+
|  C  |
+-----+
| foo |
+-----+
1 row selected (0.031 seconds)
```
tested on debian-10
```shell
./docker-hadoop.sh -d \
       --create 3 \
       --image bigtop/puppet:trunk-debian-10 \
       --memory 8g \
       --repo http://repos.bigtop.apache.org/releases/3.1.0/debian/10/\$\(ARCH\) \
       --disable-gpg-check \
       --stack hdfs,yarn,mapreduce,hbase,phoenix
./docker-hadoop.sh -e 1 /bin/bash

/# python3 /usr/lib/phoenix/bin/sqlline.py localhost
...
0: jdbc:phoenix:localhost> CREATE TABLE t(c VARCHAR PRIMARY KEY);
No rows affected (1.218 seconds)
0: jdbc:phoenix:localhost> UPSERT INTO t VALUES ('foo');
1 row affected (0.085 seconds)
0: jdbc:phoenix:localhost> SELECT * FROM t;
+-----+
|  C  |
+-----+
| foo |
+-----+
1 row selected (0.031 seconds)

```

**Comment by vivostar** [comment]
hi, @sekikn , thanks for your advice, I have updated the pr, and test it on ubuntu-18.04 and debian-10

**Comment by sekikn** [comment]
+1, tested on CentOS 7, Fedora 35 and 36, Debian 10, and Ubuntu 18.04 and 20.04.
Thanks @vivostar!

**Pull request merged**